src/services/shop/shop-router.ts

```diff
@@ -0,0 +1,275 @@
+            acc[item.itemId] = item;
+            return acc;
+        }, {});
+
+        const quantitiesMap = shopQuantities.reduce((acc: Record<string, QuantityFormat>, quantity: QuantityFormat) => {
+            acc[quantity.itemId] = quantity;
+            return acc;
+        }, {});
+
+        const itemIds = new Set([...Object.keys(itemsMap), ...Object.keys(quantitiesMap)]);
+
+        const itemsWithQuantity: ItemFormat[] = Array.from(itemIds).map((itemId: string) => {
+            const item: ShopItemFormat = itemsMap[itemId] || {
+                itemId,
+                name: "",
+                price: 0,
+                isRaffle: false,
+            };
+            const quantity: QuantityFormat = quantitiesMap[itemId] || { itemId, quantity: 0 };
+
+            return {
+                itemId,
+                name: item.name || "",
+                price: item.price || 0,
+                isRaffle: item.isRaffle || false,
+                quantity: quantity.quantity || 0,
+            };
+        });
+
+        return res.status(StatusCode.SuccessOK).send(itemsWithQuantity);
+    } catch (error) {
+        console.error(error);
+        return next(new RouterError(StatusCode.ServerErrorInternal, "InternalServerError"));
+    }
+});
+
+/**
+ * @api {post} /shop/ POST /shop/
+ * @apiGroup Shop
+ * @apiDescription Create a new item.
+ *
+ * @apiHeader {String} Authorization User's JWT Token with admin permissions.
+ *
+ * @apiBody {Json} sitem The item details to be created.
+ * @apiParamExample {Json} Request Body Example for Shop Item:
+ * {
+ *   "name": "hoodie",
+ *   "price": 15,
+ *   "isRaffle": true
+ *   "quantity": 20
+ * }
+ *
+ * @apiSuccess (201: Created) {Json} item The created item details.
+ * @apiSuccessExample Example Success Response for Shop Item
+ * HTTP/1.1 201 Created
+ * {
+ *   "item": {
+ *     "id": "52fdfc072182654f163f5f0f9a621d72",
+ *     "name": "hoodie",
+ *     "price": 15,
+ *     "isRaffle": true
+ *   }
+ *  "quantity": {
+ *      "id": "52fdfc072182654f163f5f0f9a621d72",
+ *      "quantity:" 20
+ *   }
+ * }
+ *
+ * @apiUse strongVerifyErrors
+ * @apiError (400: Bad Request) {String} InvalidParams Invalid item parameters provided.
+ * @apiError (403: Forbidden) {String} InvalidPermission User does not have admin permissions.
+ * @apiError (500: Internal Server Error) {String} InternalError An error occurred on the server.
+ */
+shopRouter.post("/", strongJwtVerification, async (req: Request, res: Response, next: NextFunction) => {
+    const payload: JwtPayload = res.locals.payload as JwtPayload;
+    // Check if the token has staff permissions
+    if (!hasStaffPerms(payload)) {
+        return next(new RouterError(StatusCode.ClientErrorForbidden, "InvalidPermission"));
+    }
+
+    // Convert item request into the item format
+    const itemFormatString: string = req.body;
+    const itemFormat: ItemFormat = JSON.parse(itemFormatString);
```

Comment:
this doesn't look right i think

Reply:
i tried doing:
```ts
const itemFormat: ItemFormat = req.body as ItemFormat; 
```
but i kept getting weird errors so i did this. is there another way to handle it?

Reply:
hm, what weird errors are you talking about?